rustfmt-core/rustfmt-lib/src/lib.rs

```diff
@@ -1,615 +1,72 @@
-use crate::utils::indent_next_line;
-
-#[macro_use]
-mod utils;
+use crate::formatting::format_input_inner;
+use crate::{emitter::Verbosity, result::OperationError};
 
 #[cfg(feature = "config")]
 pub mod config;
 #[cfg(feature = "emitter")]
 pub mod emitter;
 
-mod attr;
-mod chains;
-mod closures;
-mod comment;
-mod expr;
 mod format_report_formatter;
-pub(crate) mod formatting;
-mod ignore_path;
-mod imports;
-mod items;
-mod lists;
-mod macros;
-mod matches;
-mod missed_spans;
-pub(crate) mod modules;
-mod overflow;
-mod pairs;
-mod patterns;
+mod formatting;
 mod release_channel;
-mod reorder;
-mod rewrite;
-mod shape;
-mod skip;
-pub(crate) mod source_file;
-pub(crate) mod source_map;
-mod spanned;
-mod stmt;
-mod string;
-mod syntux;
+pub mod result;
+
 #[cfg(test)]
 mod test;
-mod types;
-mod vertical;
-pub(crate) mod visitor;
 
-/// The various errors that can occur during formatting. Note that not all of
-/// these can currently be propagated to clients.
-#[derive(Error, Debug)]
-pub enum ErrorKind {
-    /// Line has exceeded character limit (found, maximum).
-    #[error(
-        "line formatted, but exceeded maximum width \
-         (maximum: {1} (see `max_width` option), found: {0})"
-    )]
-    LineOverflow(usize, usize),
-    /// Line ends in whitespace.
-    #[error("left behind trailing whitespace")]
-    TrailingWhitespace,
-    /// License check has failed.
-    #[error("license check failed")]
-    LicenseCheck,
-    /// Used deprecated skip attribute.
-    #[error("`rustfmt_skip` is deprecated; use `rustfmt::skip`")]
-    DeprecatedAttr,
-    /// Used a rustfmt:: attribute other than skip or skip::macros.
-    #[error("invalid attribute")]
-    BadAttr,
-    /// An io error during reading or writing.
-    #[error("io error: {0}")]
-    IoError(io::Error),
-    /// Parse error occurred when parsing the input.
-    #[error("parse error")]
-    ParseError,
-    /// The user mandated a version and the current version of Rustfmt does not
-    /// satisfy that requirement.
-    #[error("version mismatch")]
-    VersionMismatch,
-    /// If we had formatted the given node, then we would have lost a comment.
-    #[error("not formatted because a comment would be lost")]
-    LostComment,
-    /// Invalid glob pattern in `ignore` configuration option.
-    #[error("Invalid glob pattern found in ignore list: {0}")]
-    InvalidGlobPattern(ignore::Error),
+/// Configures how rustfmt operates during formatting.
+#[derive(Clone, Copy, Default)]
+pub struct OperationSetting {
+    /// If set to `true`, format sub-modules which are defined in the given input.
+    pub recursive: bool,
+    pub verbosity: Verbosity,
 }
 
-impl ErrorKind {
-    fn is_comment(&self) -> bool {
-        match self {
-            ErrorKind::LostComment => true,
-            _ => false,
-        }
-    }
+/// The main entry point for Rustfmt. Formats the given input according to the
+/// given config. `out` is only necessary if required by the configuration.
```

Comment:
It's not clear to me what `out` is referencing in this context :thinking:

Reply:
This is a comment about `Session` that I forgot to update 😅 Thanks!